test/unit/specs/components/common/ShortAddress.spec.js

```diff
@@ -0,0 +1,73 @@
+import setup from "../../../helpers/vuex-setup"
+import ShortAddress from "renderer/components/common/ShortAddress"
+
+describe(`ShortAddress`, () => {
+  let wrapper
+  let instance = setup()
+
+  beforeEach(() => {
+    let test = instance.mount(ShortAddress, {
+      propsData: { address: `cosmosftw123456789` }
+    })
+    wrapper = test.wrapper
+  })
+
+  it(`has the expected html structure`, () => {
+    expect(wrapper.html()).toMatchSnapshot()
+  })
+
+  it(`should return 'address not found'`, () => {
+    wrapper.setProps({ address: null })
+    wrapper.update()
+    expect(wrapper.vm.shortAddress).toBe(`Address Not Found`)
+  })
+
+  it(`should return the address as received`, () => {
+    wrapper.setProps({ address: `add1asd` })
+    wrapper.update()
+    expect(wrapper.vm.shortAddress).toBe(`add1asd`)
+
+    wrapper.setProps({ address: `add2asd` })
+    wrapper.update()
+    expect(wrapper.vm.shortAddress).toBe(`add2asd`)
+  })
+
+  it(`should return a short address with the first 4 letters`, () => {
+    wrapper.setProps({ address: `cosmosaddress2asdfasdfasdf` })
```

Comment:
shouldn't this fail as it is not bech32?

Reply:
why `cosm…asdf` ? shouldn't it show the whole prefix + the last 4 chars ?

Reply:
I created a follow up: https://github.com/cosmos/voyager/issues/1437